Add Alpine variant

J0WI opened the pull request:
This adds an Alpine Linux variant. Shellscripts have not yet been updated.

crazy-max commented:
@J0WI An other Dockerfile.template should be created (`Dockerfile-alpine.template` for example) and rename the debain template with `Dockerfile-debian.template` and bring the modifications in `update.sh`.

J0WI commented:
@crazy-max done

pierreozoux commented on fpm-alpine/php.ini:
```diff
@@ -0,0 +1,4 @@
+always_populate_raw_post_data=-1
+extension=geoip.so
```
this is not present in other images, should we add it to others, or remove it from here?

pierreozoux commented on update.sh:
```diff
@@ -16,8 +23,9 @@ latest="$(
-for variant in apache fpm; do
-	cp Dockerfile.template "$variant/Dockerfile"
+for variant in apache fpm fpm-alpine; do
+        template="Dockerfile-${base[$variant]}.template"
```
identention is off by one space

J0WI commented:
@pierreozoux I made the requested additional changes and rebased/squashed all commits.
I also moved the `php.ini` to `conf.d/php-piwik.ini` that users have the chance to use an additional custom `php.ini`.

pierreozoux commented:
:tada:

tianon commented on Dockerfile-alpine.template:
```diff
@@ -0,0 +1,83 @@
+RUN set -ex; \
+	\
+	apk add --no-cache --virtual .build-deps \
+		alpine-sdk \
```
Doh sorry, wish I'd noticed this sooner -- this is basically the same as `build-essential` in Debian.  If you're not building Alpine APK packages, this is the wrong package to install.  You probably want instead `gcc`, `make`, etc directly instead (but also, `docker-php-ext-configure` will install those, so perhaps you wanted instead to simply include `$PHPIZE_DEPS` here?)

J0WI replied:
This seems to be required for the pecl extensions.

tianon replied:
This is more than is required, is the point I'm making -- it should be `$PHPIZE_DEPS` instead (which is the set of packages the image scripts install for doing the same thing)

J0WI replied:
Thanks, got it. This is fixed in #100.
